hooks/configure

```diff
@@ -14,119 +14,160 @@
-
-}
-
-switch_service() {
-	case "$1" in
-		false)
-			# When the unit is already enabled but not active a call with
-			# --now does start the unit so we have to check for that  case
-			# and explicitly start the unit.
-			if [ "$(systemctl is-enabled "$2.service")" = "disabled" ]; then
-				systemctl enable "$2.service"
-			fi
-			if [ "$(systemctl is-active "$2.service")" = "inactive" ]; then
-				systemctl start "$2.service"
-			fi
-			;;
-		true)
-			# A simple `systemctl disable --now ssh.service` doesn't work
-			# and fails with an error message. Because of that we're going
-			# in two steps here and disable first and then stopping the
-			# service unit.
-			systemctl disable "$2.service"
-			systemctl stop "$2.service"
-			;;
-		*)
-			echo "ERROR: Invalid value '$1' provided for option service.$2.disable"
-			exit 1
-	esac
-}
-
-update_pi_config_value() {
-    path="$1"
-    key="$2"
-    value="$3"
-
-    if [ -z "$value" ]; then
-        # empty - unset an option
-        if grep -q "^[ \t]*${key}=" "$path"; then
-            sed "/^[ \t#]*${key}=/ s/^#*/#/" "$path" > "${path}.tmp"
-            mv "${path}.tmp" "$path"
-        fi
-    else
-        if ! grep -q "^[ \t]*${key}=${value}$" "$path"; then
-            # non-empty - set an option
-            export key value
-            awk -v pat="^[ \t#]*$key=.*" --sandbox -f - "$path" >"${path}.tmp" <<'EOF'
-$0 ~ pat {
-     gsub(pat, sprintf("%s=%s",ENVIRON["key"],ENVIRON["value"]))
-     ok=1
-}
-{
-    print
-}
-END {
-    if (!ok)
-        printf("%s=%s\n",ENVIRON["key"],ENVIRON["value"])
-}
-EOF
-            mv "${path}.tmp" "$path"
-        fi
-    fi
-}
-
-for service in $SERVICES; do
-    value=$(snapctl get service.$service.disable)
-    if [ -n "$value" ]; then
-        switch_service "$value" "$service"
-    fi
-done
-
-powerkey=$(snapctl get system.power-key-action)
-if [ -n "$powerkey" ]; then
-    switch_handle_power_key "$powerkey"
-fi
-
-PI_CONFIG_KEYS="disable_overscan framebuffer_width framebuffer_height
- framebuffer_depth framebuffer_ignore_alpha overscan_left overscan_right
- overscan_top overscan_bottom overscan_scale display_rotate hdmi_group
- hdmi_mode hdmi_drive avoid_warnings gpu_mem_256 gpu_mem_512 gpu_mem"
-PI_CONFIG="${TEST_UBOOT_CONFIG:-/boot/uboot/config.txt}"
-if [ -f "$PI_CONFIG" ]; then
-    for key in $PI_CONFIG_KEYS; do
-        value=$(snapctl get "pi-config.$(echo "$key"|tr _ -)")
-        update_pi_config_value "$PI_CONFIG" "$key" "$value"
-    done
-fi
+SERVICES=("ssh",)
+
+@contextmanager
+def atomic(path):
+    """atomic is a context manager that creates a new tmpfile for the
+    given target and syncs/renames it when all writing is done, to
+    make the final operation as atomic as possible.
+    """
+    dirname, target = os.path.split(path)
+    tmp = target + ".tmp"
+    dfd = os.open(dirname, os.O_RDONLY)
+    fd = os.open(tmp, os.O_WRONLY|os.O_CREAT|os.O_EXCL, mode=0o644, dir_fd=dfd)
+
+    f = open(fd, "w", closefd=False)
```

Comment:
`os.fdopen` is an alias for `open`, and the docs for `os.fdopen` don't document `closefd` (so you need to go read the docs for `open` anyway)